### ISSUE TITLE
Unexpected Prediction TimeDistributed(BatchNormalization)

### ISSUE THREAD

**Issue by cocotdf** [keras-team/keras#16239]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Google Colab
- TensorFlow version (use command below): 2.8.0
- Python version: 3.7.12

- Exact command to reproduce:

```
import tensorflow as tf
import numpy as np
import math

timestep = 3
input_param = (timestep,5)
batch = 4
input_shape = (batch, timestep, input_param[1])


batchnorm2d = tf.keras.layers.BatchNormalization(axis = -1)

#Model with TimeDistrib(BatchNorm)
inp = tf.keras.Input(shape=(input_param))
layer = tf.keras.layers.TimeDistributed(batchnorm2d)(inp)
time_distrib_model = tf.keras.Model(inputs=(inp), outputs=(layer))

#Model with BatchNorm3D
inp_lay = tf.keras.Input(shape=(input_param))
layer = tf.keras.layers.BatchNormalization(axis=-1)(inp_lay)
batch_model = tf.keras.Model(inputs=(inp_lay), outputs=(layer))

inp = []

#Create Input
for i in range(200):
  inp.append(math.cos(i))

inp = np.resize(np.array(inp), input_shape)


#Get Predictions from the 2 models
time_distrib_pred = time_distrib_model(inp, training = True)
batch_pred = batch_model(inp, training=True)

# Display Predictions and difference
print("time_distrib predict :", time_distrib_pred)
print("batch_norm predict :", batch_pred)
print("diff between the 2 model predictions : ", (time_distrib_pred - batch_pred))
```

**Describe the problem**.

The problem is that there is no difference between TimeDistributed(BatchNormalization) and BatchNormalization.

**Describe the current behavior**.

TimeDistributed(BatchNormalization) == BatchNormalization

**Describe the expected behavior**.

If we apply a BatchNormalization layer directly, the mean and the variance would be computed for all the 3 timesteps as a whole. 
Whereas, a BatchNormalization wrapped in a TimeDistributed layer will compute the mean and variance for batches of shape ( 1 ,  5) i.e for each timestep.

**Comment by old-school-kid** [comment]
@cocotdf The [document](https://www.tensorflow.org/api_docs/python/tf/keras/layers/TimeDistributed) clearly mentions 

> Every input should be at least 3D, and the dimension of index one of the first input will be considered to be the temporal dimension

And here input means excluding batch size. But you are passing a 2D shaped tensor. Hence the error. 
Hope this helps.

**Comment by LukeWood** [comment]
Hey there @cocotdf - yeah @old-school-kid is correct.  The code is actually functioning correctly.

Thanks

**Comment by cocotdf** [comment]
@old-school-kid @LukeWood First, thanks for your answers.
I'm sorry, it's probably my fault but I don't understand, for me when we apply the BatchNormalization inside TimeDistributed, it's going to normalize every timestep according to the data of the timestep, but it's not the case, even if we place ourselves in a higher dimension like for example in 4D (with the same example but with a different dimension)

Or my understanding of TimeDistributed or BatchNormalization or even TimeDistributed(BatchNorm) is incorrect.

``````
import tensorflow as tf
import numpy as np
import math

timestep = 3
input_param = (timestep,5,6,7)
batch = 4
input_shape = (batch, timestep, 5,6,7)


batchnorm2d = tf.keras.layers.BatchNormalization(axis = -1)

#Model with TimeDistrib(BatchNorm)
inp = tf.keras.Input(shape=(input_param))
layer = tf.keras.layers.TimeDistributed(batchnorm2d)(inp)
time_distrib_model = tf.keras.Model(inputs=(inp), outputs=(layer))

#Model with BatchNorm3D
inp_lay = tf.keras.Input(shape=(input_param))
layer = tf.keras.layers.BatchNormalization(axis=-1)(inp_lay)
batch_model = tf.keras.Model(inputs=(inp_lay), outputs=(layer))

inp = []

#Create Input
for i in range(200):
  inp.append(math.cos(i))

inp = np.resize(np.array(inp), input_shape)


#Get Predictions from the 2 models
time_distrib_pred = time_distrib_model(inp, training = True)
batch_pred = batch_model(inp, training=True)

# Display Predictions and difference
print("time_distrib predict :", time_distrib_pred)
print("batch_norm predict :", batch_pred)
print("diff between the 2 model predictions : ", (time_distrib_pred - batch_pred))
``````
